resources/type-system/inference.md

```diff
@@ -1210,6 +1247,674 @@ of steps:
+String>(1, x: 2)`", this means that the input of argument part inference is
+applied to the sequence of expressions `{1, 2}`, the sequence of optional names
+`{∅, x}`, and the sequence of type arguments `{int, String}`._
+
+The procedure for argument part inference is as follows:
+
+- Let `{X_1, X_2, ...}` be the list of formal type parameters in `F`, or an
+  empty list if `F` is `∅`.
+
+- Let `{P_1, P_2, ...}` be the list of formal parameter types corresponding to
+  `{e_1, e_2, ...}`, determined as follows:
+
+  - If `F` is `∅`, then let each `P_i` be `_`.
+
+  - Otherwise:
+
+    - For each `n_i` that is `∅`, let `P_i` be the type of the corresponding
+      positional parameter in `F`. If there is no such parameter, it is a
+      compile time error (_too many positional arguments supplied_).
+
+    - For each `n_i` that is not `∅`, let `P_i` be the type of the parameter of
+      `F` named `n_i`. If there is no such parameter, it is a compile time error
+      (_unnexpected named argument supplied_).
+
+- Let `R_F` be the return type of `F`. Or, if `F` is `∅`, then let `R_F` be
+  `dynamic`.
+
+- Produce an initial list of type constraints `C`, and an initial list of type
+  schemas `{U_1, U_2, ...}`, as follows:
+
+  - If `F` is `∅`, then:
+
+    - Let `C` be an empty list of type constraints, and let `{U_1, U_2, ...}` be
+      the same as `{T_1, T_2, ...}`. _This covers the case of dynamic
+      invocation; the user-supplied type arguments are accepted without
+      modification._
+
+  - Otherwise, `F` must be a function type. If the number of formal type
+    parameters of `F` (which could be zero) matches the number of type arguments
+    `T`, then:
+
+    - Let `C` be an empty list of type constraints, and let `{U_1, U_2, ...}` be
+      the same as `{T_1, T_2, ...}`. _This covers the case where no generic type
+      inference is needed, because a correct number of explicit type arguments
+      was supplied._
+
+  - Otherwise, if the number of type arguments `T` is nonzero, then there is a
+    compile-time error (_wrong number of type arguments supplied_).
+
+  - Otherwise, `F` must have at least one formal type parameter, and there must
+    be zero type arguments `T` (_in other words, type arguments will be
+    inferred_). Then:
+
+    - Initialize `C` to an empty list of type constraints.
+
+    - Initialize `{U_1, U_2, ...}` to a list of type schemas, of the same length
+      as `{X_1, X_2, ...}`, each of which is `_`.
+
+    - Using subtype constraint generation, attempt to match `R_F` as a subtype
+      of `K` with respect to the list of type variables `{X_1, X_2, ...}`.
+
+      - If this succeeds, then accumulate the resulting list of type constraints
+        into `C`. Then, let `{V_1, V_2, ...}` be the constraint solution for the
+        set of type variables `{X_1, X_2, ...}` with respect to the constaint
+        set `C`, with partial solution `{U_1, U_2, ...}`. Replace `{U_1, U_2,
```

Comment:
Maybe say "Let the new values of `{U_1, ...}` be `{V_1...}`?  Saying "replace" is a little confusing here.

Reply:
Done (here and elsewhere in the PR).